server/render/index.js

```diff
@@ -226,3 +199,57 @@ export function serverRenderError( err, req, res, next ) {
 
 	next();
 }
+
+/**
+ * A generic middleware that identifies if the current request fits the conditions for being SSRable.
+ *
+ * Warning: Having serverSideRender=true is not sufficient. The app-level logic needs be applied before truly SSRing.
+ *
+ * @param {object}   context  The entire request context
+ * @param {function} next     As all middlewares, will call next in the sequence
+ */
+export function setShouldServerSideRender( context, next ) {
+	context.serverSideRender = Object.keys( context.query ).length === 0; // 0 query parameters expected
```

Comment:
I think both the JSDoc and this comment could be a bit more explanatory:

* The JSDoc should maybe make clear that this is a middleware that's run early in the chain, and that sets a fallback for whether to SSR or not, which other middleware (run later in the chain) are allowed to override.
* The inline comment could explain the implementation detail that generally we don't want to SSR if there are any query args.